js/filters.js

```diff
@@ -115,9 +145,27 @@ function TypeaheadFilter(elm) {
   var key = this.$body.data('dataset');
   var dataset = typeahead.datasets[key];
   this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
```

Comment:
Come to think of it, why is this setup this way as opposed to the way the other objects are setup?  Is it because the Typeahead filter itself is being defined in here with the other filter types, but its behavior is being defined in `typeahead-filter.js`?